exercises/practice/leap/.approaches/introduction.md

```diff
@@ -0,0 +1,83 @@
+# Introduction
+
+Every fourth year is a leap year (with some exceptions), but let's consider this one condition first. 
+
+To solve the Leap problem, we must determine if a year is evenly divisible by a number or if a reminder of an integer division is zero. 
+Such operation in computing is called [modulo][modulo]. 
+
+Unlike many languages, Elixir does not have [operators][operators] for either integer division or modulo. 
+Instead, it provides the [`Kernel.rem/2`][rem] and the [`Integer.mod/2`][mod] functions.
+
+The two functions differ in how they work with negative numbers, but since, in this exercise,
+all the numbers are non-negative, both could work, depending on the approach.
+
+## General solution
+
+To check if a year is divisible by `n`, we can do `rem(year, n) == 0`. 
+
+Any approach to the problem will perform this check three times to see if a year is equally divisible by 4, 100 and 400.
+What will differ between approaches is what Elixir features we will use to combine the checks. 
+
+## Approach: Boolean operators
+
+The full rules are as follows:
+A year is a leap year if 
+* it is divisible by 4 
+* but not divisible by 100
+* unless it is divisible by 400
+
+We can use [boolean operators][boolean-operators] to combine the checks, for example, like so: 
+
+```elixir
+rem(year, 5) == 0 and not rem(year, 100) == 0 or rem(year, 400) == 0
+```
+In the [boolean operators appraoch][operators-approach] we discuss the details of the solution.
+It includes variations of the operators and their precendence.
+
+## Approach: multiple clause function
+
+Instead of using boolean operators, we can define multiple `leap_year?/1` function clauses with different guards. 
+
+```elixir
+def leap_year?(year) when rem(year, 400) == 0, do: true
+def leap_year?(year) when rem(year, 100) == 0, do: false
+def leap_year?(year) when rem(year, 4) == 0, do: true
+def leap_year?(_), do: false
+```
+
+In the [multiple clause functions approach][functions-approach] we discuss why in this approach the `Integer.mod/2` function will not work. 
```

Comment:
There is only one function in this exercise, `leap_year?`:
```suggestion
In the [multiple clause function approach][functions-approach] we discuss why in this approach the `Integer.mod/2` function will not work. 
```

Also, the directory name of this concept, "functions", and this link name, `[functions-approach]` (also used in other markdown files), is not appropriate. Every single approach has "functions" 😁 this name should be something distinctive about the approach. Maybe "multiple-clause-function", or shorter, "multiple-clauses"

Reply:
@angelikatyborska, how about `clauses` or `function-clauses` for the slug?

Reply:
sounds good too